### PR TITLE
Minor refactoring of HivePageSourceProvider

### PR DESCRIPTION
**Pull request by sdruzkin**
## Description
Do minor refactoring of HivePageSourceProvider by extracting a few variables here and there, 
and moving some variables out of the loops.

## Motivation and Context
Make the code a bit better.

## Impact
No impact

## Test Plan
Existing tests

## Contributor checklist

- [x] Please make sure your submission complies with our [development](https://github.com/prestodb/presto/wiki/Presto-Development-Guidelines#development), [formatting](https://github.com/prestodb/presto/wiki/Presto-Development-Guidelines#formatting), [commit message](https://github.com/prestodb/presto/wiki/Review-and-Commit-guidelines#commit-formatting-and-pull-requests), and [attribution guidelines](https://github.com/prestodb/presto/wiki/Review-and-Commit-guidelines#attribution).
- [x] PR description addresses the issue accurately and concisely.  If the change is non-trivial, a GitHub Issue is referenced.
- [x] Documented new properties (with its default value), SQL syntax, functions, or other functionality.
- [x] If release notes are required, they follow the [release notes guidelines](https://github.com/prestodb/presto/wiki/Release-Notes-Guidelines).
- [x] Adequate tests were added if applicable.
- [x] CI passed.

## Release Notes
Please follow [release notes guidelines](https://github.com/prestodb/presto/wiki/Release-Notes-Guidelines) and fill in the release notes below.

```
== NO RELEASE NOTE ==
```

### CODE REVIEW

File: presto-hive/src/main/java/com/facebook/presto/hive/HivePageSourceProvider.java [hunk 1]
```diff
@@ -255,16 +255,17 @@ private ConnectorPageSource createAggregatedPageSource(
...
     {
-        for (HiveAggregatedPageSourceFactory pageSourceFactory : aggregatedPageSourceFactories) {
-            List<ColumnMapping> columnMappings = ColumnMapping.buildColumnMappings(
-                    hiveSplit.getPartitionKeys(),
-                    selectedColumns,
-                    hiveSplit.getBucketConversion().map(BucketConversion::getBucketColumnHandles).orElse(ImmutableList.of()),
-                    hiveSplit.getTableToPartitionMapping(),
-                    hiveSplit.getFileSplit(),
-                    hiveSplit.getTableBucketNumber());
+        List<ColumnMapping> columnMappings = ColumnMapping.buildColumnMappings(
```
Comment (sdruzkin) [inline presto-hive/src/main/java/com/facebook/presto/hive/HivePageSourceProvider.java:258]: Moved this variable out of the loop.

File: presto-hive/src/main/java/com/facebook/presto/hive/HivePageSourceProvider.java [hunk 2]
```diff
@@ -364,6 +365,10 @@ private static Optional<ConnectorPageSource> createSelectivePageSource(
             return Optional.of(new HiveEmptySplitPageSource());
         }
 
+        TupleDomain<Subfield> domainPredicate = splitContext.getDynamicFilterPredicate()
```
Comment (sdruzkin) [inline presto-hive/src/main/java/com/facebook/presto/hive/HivePageSourceProvider.java:368]: Extracted this really long line into its own variable.

File: presto-hive/src/main/java/com/facebook/presto/hive/HivePageSourceProvider.java [hunk 3]
```diff
@@ -549,7 +551,6 @@ private static Optional<ConnectorPageSource> getPageSourceFromCursorProvider(
             HiveFileSplit fileSplit,
             Storage storage,
             TupleDomain<HiveColumnHandle> effectivePredicate,
-            List<HiveColumnHandle> hiveColumns,
```
Comment (sdruzkin) [inline presto-hive/src/main/java/com/facebook/presto/hive/HivePageSourceProvider.java:552]: Removed unused argument.

**Comment by rschlussel** [comment]
Thank you!

**Pull request merged**